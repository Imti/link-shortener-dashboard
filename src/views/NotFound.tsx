import { useNavigate } from 'react-router-dom'
import { FaArrowLeft } from 'react-icons/fa'

export default function NotFound() {
  const navigate = useNavigate()
  return (
    <div>
      <span
        className="flex items-center text-left text-sm text-slate-400 mb-4 p-2 cursor-pointer hover:underline"
        onClick={() => navigate('/')}
      >
        <FaArrowLeft size="10px" className="mr-1" />
        back to all links
      </span>
      <span className="block text-xl md:text-2xl font-semibold text-slate-300 mb-1">
        Page Not Found
      </span>
    </div>
  )
}
