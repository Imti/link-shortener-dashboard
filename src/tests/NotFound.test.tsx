import renderer from 'react-test-renderer'
import { BrowserRouter as Router } from 'react-router-dom'
import NotFound from '../views/NotFound'

describe('<NotFound />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Router>
          <NotFound />
        </Router>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
