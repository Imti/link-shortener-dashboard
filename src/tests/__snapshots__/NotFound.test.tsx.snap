// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> renders correctly 1`] = `
<div>
  <span
    className="flex items-center text-left text-sm text-slate-400 mb-4 p-2 cursor-pointer hover:underline"
    onClick={[Function]}
  >
    <svg
      className="mr-1"
      fill="currentColor"
      height="10px"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 448 512"
      width="10px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z"
      />
    </svg>
    back to all links
  </span>
  <span
    className="block text-xl md:text-2xl font-semibold text-slate-300 mb-1"
  >
    Page Not Found
  </span>
</div>
`;
